test-perf/GeneratorBenchmark.ts

```diff
@@ -0,0 +1,86 @@
+import "mocha";
+import * as assert from "power-assert";
+
+import { ActionResult, successOn } from "../src/action/ActionResult";
+import { GitHubRepoRef } from "../src/operations/common/GitHubRepoRef";
+import { ProjectOperationCredentials } from "../src/operations/common/ProjectOperationCredentials";
+import { RepoId } from "../src/operations/common/RepoId";
+import { generate, ProjectPersister } from "../src/operations/generate/generatorUtils";
+import { LocalProject } from "../src/project/local/LocalProject";
+import { NodeFsLocalProject } from "../src/project/local/NodeFsLocalProject";
+import { InMemoryProject } from "../src/project/mem/InMemoryProject";
+import { Project } from "../src/project/Project";
+
+describe("generator benchmark", () => {
+
+    const numberOfTimesToGenerate = 500;
+    const numberOfSeedProjectFiles = 100;
+    const numberOfLipsumInFile = 10;
+
+    it("should create a new GitHub repo using GenericGenerator", done => {
+        const repoName = "perfTestRepo";
+
+        const seed = constructProject(numberOfSeedProjectFiles, numberOfLipsumInFile);
+        generateFromSeed(seed, repoName)
+        .then(r => {
+            const result = r as ActionResult<LocalProject>;
+            NodeFsLocalProject.fromExistingDirectory(new GitHubRepoRef("atomist", repoName), result.target.baseDir)
+            .then(created => {
+                created.totalFileCount().then(fileCount =>
+                    assert.deepEqual(fileCount, numberOfSeedProjectFiles),
+                ).then(
+                    () => {
+                        let lastGeneration: Promise<any>;
+                        for (let i = 1; i < numberOfTimesToGenerate ; i++) {
+                            lastGeneration = generateFromSeed(seed, repoName);
+                        }
+                        if (lastGeneration) {
+                            lastGeneration.then(() => done());
```

Comment:
Should this be using a `Promise.all` approach? The last generation could finish before all the rest, right?

Reply:
probably in real production code where it really matters. It even works fine if done happens immediately after the assertion, it just looks weird because the test is still going.

Reply:
That should probably be fixed then, right?